docs/cookbook/adding-new-attribute-to-an-entity.md

```diff
@@ -0,0 +1,361 @@
+
+class ProductFacade extends BaseProductFacade
+{
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\ProductEditData $productEditData
+     * @return \Shopsys\ShopBundle\Model\Product\Product
+     */
+    public function create(ProductEditData $productEditData)
+    {
+        $product = Product::create($productEditData->productData);
+
+        // ... (copy paste from BaseProductFacade)
+
+        return $product;
+    }
+}
+```
+
+9. Create new `ProductDataFactory` in the same namespace as your entity
+by extending [`Shopsys\FrameworkBundle\Model\Product\ProductDataFactory`](../../packages/framework/src/Model/Product/ProductDataFactory.php)
+and overwrite the `createDefault()` method. 
+You need to copy paste the contents of the parent class and make sure to use your `Product` instead of the base one.
+```php
+<?php
+
+namespace Shopsys\ShopBundle\Model\Product;
+
+use Shopsys\FrameworkBundle\Model\Pricing\Vat\VatFacade;
+use Shopsys\FrameworkBundle\Model\Product\Pricing\ProductInputPriceFacade;
+use Shopsys\FrameworkBundle\Model\Product\ProductDataFactory as BaseProductDataFactory;
+use Shopsys\FrameworkBundle\Model\Product\Unit\UnitFacade;
+
+class ProductDataFactory extends BaseProductDataFactory
+{
+    // ... (copy paste from BaseProductDataFactory)
+
+    /**
+     * @return \Shopsys\ShopBundle\Model\Product\ProductData
+     */
+    public function createDefault()
+    {
+        $productData = new ProductData();
+
+        // ... (copy paste from BaseProductDataFactory)
+
+        return $productData;
+    }
+}
+```
+
+10. Create new `ProductVariantService` in the same namespace as your entity
+by extending [`Shopsys\FrameworkBundle\Model\Product\ProductVariantService`](../../packages/framework/src/Model/Product/ProductVariantService.php)
+and overwrite the `createMainVariant()` method.
+Make sure to use your `Product` instead of the base one. 
+```php
+<?php
+
+namespace Shopsys\ShopBundle\Model\Product;
+
+use Shopsys\FrameworkBundle\Model\Product\Product as BaseProduct;
+use Shopsys\FrameworkBundle\Model\Product\ProductEditData as BaseProductEditData;
+use Shopsys\FrameworkBundle\Model\Product\ProductVariantService as BaseProductVariantService;
+
+class ProductVariantService extends BaseProductVariantService
+{
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\ProductEditData $mainVariantEditData
+     * @param \Shopsys\ShopBundle\Model\Product\Product $mainProduct
+     * @param \Shopsys\ShopBundle\Model\Product\Product[] $variants
+     * @return \Shopsys\ShopBundle\Model\Product\Product
+     */
+    public function createMainVariant(BaseProductEditData $mainVariantEditData, BaseProduct $mainProduct, array $variants)
+    {
+        $variants[] = $mainProduct;
+
+        return Product::createMainVariant($mainVariantEditData->productData, $variants);
+    }
+}
+``` 
+
+11. Set your new `ProductFacade`, `ProductDataFactory` and `ProductVariantService`
```

Comment:
Just idea to discuss, would not be better if every service was added right when it is created?

Reply:
yes, you are right :+1: